### PR TITLE
Add 3.12 wheels to build

### PR DESCRIPTION
**Pull request by hmstepanek**
# Overview
This PR fixes up some issues with supporting 3.12.
* Add 3.12 into the list of wheel build envs.

### CODE REVIEW

File: .github/containers/Dockerfile [hunk 1]
```diff
@@ -96,12 +96,12 @@ RUN echo 'eval "$(pyenv init -)"' >>${HOME}/.bashrc && \
     pyenv update
 
 # Install Python
-ARG PYTHON_VERSIONS="3.11 3.10 3.9 3.8 3.7 3.12 2.7 pypy2.7-7.3.12 pypy3.8-7.3.11"
+ARG PYTHON_VERSIONS="3.12 3.11 3.10 3.9 3.8 3.7 3.12 2.7 pypy2.7-7.3.12 pypy3.8-7.3.11"
```
Comment (TimPansino) [inline .github/containers/Dockerfile:99]:
```suggestion
ARG PYTHON_VERSIONS="3.11 3.10 3.9 3.8 3.7 3.12 2.7 pypy2.7-7.3.12 pypy3.8-7.3.11"
```

3.12 is already in the versions list, it's at the end because we usually use a more stable version as our default for running tools such as tox.

**Review by TimPansino — CHANGES_REQUESTED** [review]
One change and this looks good

File: newrelic/packages/isort/stdlibs/py312.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
+"""
+File contains the standard library of Python 3.12.
+
+DO NOT EDIT. If the standard library changes, a new list should be created
+using the mkstdlibs.py script.
+"""
+
```
Comment (TimPansino) [inline newrelic/packages/isort/stdlibs/py312.py:7]: Is this file necessary at all? Based on the below code I don't think we ever reference it.

```python
    elif python_version >= (3, 10):
        stdlibs = sys.stdlib_module_names
```

Comment (TimPansino) [reply newrelic/packages/isort/stdlibs/py312.py:7]: It really doesn't require python 3.10 or 3.11 either, but those were probably just copied in since they were there. I'm unsure if there's any reason to update these.

File: .github/containers/Dockerfile (hunk 1 above)
Comment (hmstepanek) [reply .github/containers/Dockerfile:99]: oh no! I see it now oops.

File: newrelic/packages/isort/stdlibs/py312.py (hunk 2 above)
Comment (hmstepanek) [reply newrelic/packages/isort/stdlibs/py312.py:7]: Ah true. It looks like there were only changes to 3.11 and 3.12 here. Given that both those aren't used, I guess we can just skip upgrading isort for now.

**Review by TimPansino — APPROVED** [review]

File: .github/workflows/deploy-python.yml [hunk 3]
```diff
@@ -57,7 +57,7 @@ jobs:
         uses: pypa/cibuildwheel@v2.11.1
```
Comment (TimPansino) [inline .github/workflows/deploy-python.yml:57]:
```suggestion
        uses: pypa/cibuildwheel@v2.16.2
```

**Review by TimPansino — CHANGES_REQUESTED** [review]
This needs to be updated or it won't have 3.12 support and the CI will crash.

**Comment by TimPansino** [comment]
Reposting this comment

> Closing this since this PR does it and more: https://github.com/newrelic/newrelic-python-agent/pull/1024.
